### ISSUE TITLE
[Method feasibility]: Whether to support the use of pandas module python to convert into *.pmml file

### ISSUE THREAD

**Issue by yijiantao** [jpmml/jpmml-python#13]
Hello,
I'm trying to use jpmml/jpmml-python and I would like to ask a question about how to use java to implement a function based on the pandas module. 

The function using pandas.DataFrame, like this:
```python
import pandas as pd
def psi(data_dict):
    data = {'actucal':data_dict['actucal'].values(), 'expected':data_dict['expected'].values()}
    df = pd.DataFrame(data, index = data_dict['actucal'].keys())
    df['ind'] = (df['actucal'] - df['expected']) * numpy.log(df['actucal'] / df['expected'])
    psi = sum(df['ind'])
    return psi
```

Perhaps, can I use the command-line to convert? like this:
```shell
java -jar target/jpmml-python-executable-*.jar  #--{some_parameters}  file_name.pkl  --pmml-output file_name.pmml
```

Thank you in advance for taking the time to answer my question!
@vruusmann

**Comment by vruusmann** [comment]
> data = {'actucal':data_dict['actucal'].values(), 'expected':data_dict['expected'].values()}
> df = pd.DataFrame(data, index = data_dict['actucal'].keys())

The PMML language does not provide means for working with "data containers" such as matrix-like data tables, dict-like data tables etc. You need to keep this part of the workflow outside of the (J)PMML workflow.

> df['ind'] = (df['actucal'] - df['expected']) * numpy.log(df['actucal'] / df['expected'])

Assuming you have data loaded into a `DataFrame` that has `"actual"` and `"expected"` columns:

```python
from sklearn2pmml.preprocessing import ExpressionTransformer

# The TransformerMixin.fit(X) method exposes the pandas.DataFrame object to Python expression as "X" variable.
transformer = ExpressionTransformer("(X['actual'] - X['expected']) * numpy.log(X['actual'] / X['expected'])")
```

> psi = sum(df['ind'])

Do you mean summing all rows in the data table?

If so, then this cannot be done, because the PMML language operates on single (isolated) data records.

TLDR: It looks to me that your application would be better served by SQL (not PMML).
